demisto_sdk/commands/common/docker/dockerhub_client.py

```diff
@@ -0,0 +1,395 @@
+import os
+from datetime import datetime, timedelta
+from enum import Enum
+from functools import lru_cache
+from typing import Any, Dict, List, Optional
+
+import dateparser
+import requests
+from packaging.version import InvalidVersion, Version
+from requests.exceptions import ConnectionError, HTTPError, Timeout
+
+from demisto_sdk.commands.common.handlers.xsoar_handler import JSONDecodeError
+from demisto_sdk.commands.common.logger import logger
+from demisto_sdk.commands.common.tools import retry
+
+DOCKERHUB_USER = "DOCKERHUB_USER"
+DOCKERHUB_PASSWORD = "DOCKERHUB_PASSWORD"
+DEFAULT_REPOSITORY = "demisto"
+
+
+class DockerHubAuthScope(str, Enum):
+    PULL = "pull"  # Grants read-only access to the repository, allowing you to pull images.
+    PUSH = "push"  # Grants write access to the repository, allowing you to push images.
+    DELETE = "delete"  # Grants permission to delete images from the repository.
+    REPOSITORY = "repository"  # Grants full access to the repository, including both pull and push permissions.
+
+
+class DockerHubClient:
+
+    DEFAULT_REGISTRY = "https://registry-1.docker.io/v2"
+    DOCKER_HUB_API_BASE_URL = "https://hub.docker.com/v2"
+
+    def __init__(
+        self,
+        docker_hub_api_url: str = "",
+        registry: str = "",
+        username: str = "",
+        password: str = "",
+        verify_ssl: bool = False,
+    ):
+
+        self.registry_api_url = registry or self.DEFAULT_REGISTRY
```

Comment:
can we support an env variable for this? We should support the same ones the docker client supports

Reply:
i don't see why not